rimage/lazy_encoded.go

```diff
@@ -29,6 +32,11 @@ type LazyEncodedImage struct {
 // away with reading all metadata from the header of the image bytes.
 // NOTE: Usage of an image that would fail to decode causes a lazy panic.
 func NewLazyEncodedImage(imgBytes []byte, mimeType string) image.Image {
+	if mimeType == "" {
+		logging.Global().Warn("NewLazyEncodedImage called without a mime_type. " +
+			"Sniffing bytes to detect mime_type. Specify mime_type to reduce CPU utilization")
+		mimeType = http.DetectContentType(imgBytes)
```

Comment:
This will attempt to detect the mime type of the image bytes if mimeType is empty.

Reply:
This is a cool function from that library - nice to know it's in there.

Reply:
It [seems this method](https://pkg.go.dev/net/http#DetectContentType) can return 3 mime types that would be interesting to us:
* image/*
* application/octet-stream
* something else

It might be worth logging if this code path is getting application/octet-stream
Definitely worth logging if it's "something else"

Reply:
great suggestion, done